### ISSUE TITLE
Respawn prompts vanish upon stopping spectating a player

### ISSUE THREAD

**Issue by unfunnyperson** [goonstation/goonstation#13996]
### Labels

[BUG]

### Description

If you stop spectating a player while having a respawn prompt open the prompt vanishes

### Steps to reproduce

1. Die
2. Spectate a player
3. Receive a respawn prompt while spectating the player
4. Stop spectating the player and have your hopes of respawning crushed

### Expected Behavior

The prompt to not vanish

**Comment by frawhst** [comment]
Yeah tgui stuff is linked to a mob usually, needs to be transferred manually afaik
